### PR TITLE
fix #205 Negative literals

### LINKED ISSUE #205: Negative literals seem to be unsupported
I have found that negative const literals are not supported. It seems like they should be.
I made a simple rust crate to demonstrate:
```rust
const POS_ONE: i8 = 1;
const NEG_ONE: i8 = -1;

#[no_mangle]
pub extern fn hello() {
    println!("POS_ONE is {}, NEG_ONE is {}", POS_ONE, NEG_ONE);
}
```

Here are the command lines I used and the warning I got:
```
C:\projects>cargo new cbind_neg_lit --lib
     Created library `cbind_neg_lit` project

C:\projects>code cbind_neg_lit

C:\projects>cd cbindgen

C:\projects\cbindgen>cargo run ..\cbind_neg_lit -o ..\cbind_neg_lit\cbind_neg_lit.h
    Finished dev [unoptimized + debuginfo] target(s) in 0.20s
     Running `target\debug\cbindgen.exe ..\cbind_neg_lit -o ..\cbind_neg_lit\cbind_neg_lit.h`
WARN: Skip cbind_neg_lit::NEG_ONE - (Unsupported literal expression.)

C:\projects\cbindgen>
```
And here is the file that was generated:
```c
#include <cstdint>
#include <cstdlib>

static const int8_t POS_ONE = 1;

extern "C" {

void hello();

} // extern "C"
```

### PR DESCRIPTION
**Pull request by AxelNennker**
Signed-off-by: Axel Nennker <axel.nennker@telekom.de>

Fix https://github.com/eqrion/cbindgen/issues/205

### CODE REVIEW

File: src/bindgen/ir/constant.rs [hunk 1]
```diff
@@ -209,7 +210,19 @@ impl Literal {
...
                 })
             }
-            _ => Err("Unsupported literal expression.".to_owned()),
+            // Unary(ExprUnary { attrs: [], op: Neg(Sub), expr: Lit(ExprLit { attrs: [], lit: Int(LitInt { token: Literal { lit: 1 } }) }) }))
+            syn::Expr::Unary(syn::ExprUnary {
+                attrs: _,
+                ref op,
+                ref expr,
+            }) => match *op {
+                UnOp::Neg(_) => {
```
Comment (RReverser) [inline src/bindgen/ir/constant.rs]: While at it, maybe worth adding `!` as well? Seems easy enough to add another branch.

Comment (AxelNennker) [reply src/bindgen/ir/constant.rs]: Well, I tried to keep it simple as I did this in my free time on a week end.
Negative integer constants seem to be common while `pub const FALSE: bool = !true;` looks silly.
I would wait for somebody to have this problem.
There are other things I would like cbindgen being able to handle like 
`WARN: Skip constant::BAR - (Unhandled const definition)` from 
`const BAR: &'static str = "hello world";` which is very common, I think.

Comment (RReverser) [reply src/bindgen/ir/constant.rs]: Fair enough :) Sorry for derailing the PR.

File: src/bindgen/ir/constant.rs [hunk 2]
```diff
@@ -209,7 +210,19 @@ impl Literal {
                     fields: field_pairs,
                 })
             }
-            _ => Err("Unsupported literal expression.".to_owned()),
+            // Unary(ExprUnary { attrs: [], op: Neg(Sub), expr: Lit(ExprLit { attrs: [], lit: Int(LitInt { token: Literal { lit: 1 } }) }) }))
```
Comment (emilio) [inline src/bindgen/ir/constant.rs]: I'd remove this comment.

**Pull request merged**